sycl/test-e2e/format.py

```diff
@@ -234,15 +234,21 @@ def get_extra_env(sycl_devices):
                 new_script.append(directive)
                 continue
 
-            # Filter commands based on split-mode
+            # Filter commands based on testing mode
             is_run_line = any(
                 i in directive.command
                 for i in ["%{run}", "%{run-unfiltered-devices}", "%if run-mode"]
             )
 
-            if (is_run_line and test.config.test_mode == "build-only") or (
-                not is_run_line and test.config.test_mode == "run-only"
-            ):
+            ignore_line_filtering = (
+                "build-and-run-mode" in test.requires
+                and test.config.test_mode == "run-only"
```

Comment:
Is `and test.config.test_mode == "run-only"` here redundant? Because `and test.config.fallback_build_run_only` will ONLY be `True` when test mode is run-only.

Reply:
True, let me go fix that